File: BTS-SIO/CSharp2/2ème année/TD/SLAM45_TPTDHeritageConstructeur_vehicule_Voiture Eleve/LibVehicule/Voiture.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SLAM45_TPTDHeritageConstructeur
{
    public class Voiture : Vehicule
    {
        private int nbVitesse;
        private string typeBoite;

        public Voiture(string immat, string couleur, string typeBoite, int nbVitesse)
            :base(immat, couleur)
        {
            this.typeBoite = typeBoite;
            this.nbVitesse = nbVitesse;
        }
        public Voiture(string immat, string typeBoite,int nbVitesse)
            :base(immat)
        {
            this.typeBoite = typeBoite;
            this.nbVitesse = nbVitesse;
        }
        public Voiture(string immat, int nbVitesse)
            :base(immat)
        {
            this.nbVitesse = nbVitesse;
        }

        public override string ToString()
        {
            return base.ToString()+ nbVitesse+"/"+typeBoite;
        }
    }
}

File: BTS-SIO/Java/SLAM45_TPTDHeritageConstructeur/src/main/java/com/mycompany/slam45_tptdheritageconstructeur/Voiture.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.slam45_tptdheritageconstructeur;

/**
 *
 * @author Thomas
 */
public class Voiture extends Vehicule {
    private int nbVitesse;
    private String typeBoite;

    Voiture(String immat, String couleur, String typeBoite, int nbVitesse) {
        super(immat, couleur);
        this.typeBoite = typeBoite;
        this.nbVitesse = nbVitesse;
    }
    Voiture(String immat, String typeBoite,int nbVitesse) {
        super(immat);
        this.typeBoite = typeBoite;
        this.nbVitesse = nbVitesse;
    }
    Voiture(String immat, int nbVitesse) {
        super(immat);
        this.nbVitesse = nbVitesse;
    }

    @Override
    public String toString() {
        return super.toString()+ nbVitesse+"/"+typeBoite;
    }
}